### PR TITLE
DDBTEAM-447: Added custom populate command

### PR DESCRIPTION
**Pull request by tuj**

### CODE REVIEW

File: src/Repository/SearchRepository.php [hunk 1]
```diff
@@ -18,4 +18,38 @@ public function __construct(RegistryInterface $registry)
...
+     * @return \Doctrine\ORM\Query
+     */
+    public function findByIdRangeQuery(int $startId, int $endId)
+    {
+        $queryBuilder = $this->createQueryBuilder('e');
+        $queryBuilder
+            ->andWhere('e.id >= :startId')
+            ->andWhere('e.id < :endId')
+            ->setParameter('startId', $startId)
+            ->setParameter('endId', $endId)
```
Comment (turegjorup) [inline src/Repository/SearchRepository.php]: Would prefer the use of `$queryBuilder->setMaxResults($limit)`

File: src/Command/SearchPopulateCommand.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            throw new \RuntimeException('Index must be specified.');
+        }
+
+        $progressBar = new ProgressBar($output);
+        $progressBar->setFormat('[%bar%] %elapsed% (%memory%) - %message%');
+
+        $this->populateService->setProgressBar($progressBar);
+        $this->populateService->populate($index);
+
+        $style->success('Successfully populated.');
```
Comment (turegjorup) [inline src/Command/SearchPopulateCommand.php]: You already called 'finish' on the Progressbar at this point. I think all progress messages should be kept in the `populateService`

File: src/Repository/SearchRepository.php [hunk 3]
```diff
@@ -14,8 +18,42 @@
...
+        $lastEntity = $this->findOneBy([], ['id' => 'DESC']);
+
+        return $lastEntity->getId();
+    }
+
+    /**
+     * Get number of records.
+     *
+     * @return int
+     *
```
Comment (cableman) [inline src/Repository/SearchRepository.php]: Missing description on return statement

File: src/Service/PopulateService.php [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+     *   The Search repository
+     * @param \Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface $parameterBag
+     *   The parameter bag
+     */
+    public function __construct(EntityManagerInterface $entityManager, SearchRepository $searchRepository, ParameterBagInterface $parameterBag)
+    {
+        $this->searchRepository = $searchRepository;
+        $this->entityManager = $entityManager;
+        $this->elasticHost = $parameterBag->get('elastic.url');
+        $entityManager->getConnection()->getConfiguration()->setSQLLogger(null);
```
Comment (cableman) [inline src/Service/PopulateService.php]: A comment about why this is need and what it do would be nice?

File: src/Service/PopulateService.php [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        $lastId = $this->searchRepository->findLastId();
+        $currentId = 0;
+
+        while ($entriesAdded < $numberOfRecords) {
+            $entities = $this->searchRepository->findBy([], ['id' => 'ASC'], self::BATCH_SIZE, $entriesAdded);
+
+            // No more results.
+            if (0 === count($entities)) {
+                $this->progressMessage(sprintf('%d of %d processed. Id: %d. Last id: %d. No more results.', $entriesAdded, $numberOfRecords, $currentId, $lastId));
+                $this->progressAdvance();
```
Comment (cableman) [inline src/Service/PopulateService.php]: Why advance, when there is no more results?

File: src/Repository/SearchRepository.php [hunk 6]
```diff
@@ -32,18 +32,20 @@ public function __construct(RegistryInterface $registry)
      * Find the last id.
      *
      * @return int|null
+     *   The last id or null
      */
     public function findLastId()
```
Comment (cableman) [inline src/Repository/SearchRepository.php]: You could add `: ?int` to the function declaration.

**Pull request merged**